src/software/world/game_state.h

```diff
@@ -357,4 +345,23 @@ class GameState
      *         in a state of play where the ball needs to be placed
      */
     std::optional<Point> getBallPlacementPoint() const;
+
+    State getState() const;
+
+    void setState(State state);
+    void setRestartReason(RestartReason restart_reason);
+    void setOurRestart(bool our_restart);
+
+   private:
+    State state_;
```

Comment:
can we give State a more specific name, e.g. PlayState?

Reply:
changed to PlayState